### ISSUE TITLE
RealmQuery findAllAsync() not working

### ISSUE THREAD

**Issue by idok595** [realm/realm-java#4626]
realm version:  3.1.4
i am trying to RealmQuery.findAllAsync after creating a realm insatanse,my table is very big arround 10K records + 10-20MB, some times the RealmChangeListener called and sometimes not, on debug mode in 90% of times it called, but in normal mode it dosent called at all.

**Comment by Zhuinden** [comment]
You need to store the results as a field variable

**Comment by cmelchior** [comment]
Note you need to keep a reference to your `RealmResults` to avoid it being GC'ed. Most likely in a class variable:

```
public Foo {
  private RealmResults<Bar> results;

  private baz() {
    results = realm.where(Bar.class).findAllAsync();
    results.addChangeListener(listener);
  }
}
```

If that is not your problem, you need to provide some code we can look at.

**Comment by idok595** [comment]
yes that is my problem, but why?, by the way there is nothing on the documentation about it...

**Comment by cmelchior** [comment]
It is on the website: https://realm.io/docs/java/latest/#notifications
But we should probably also make it more clear in the Javadoc.

**Comment by cmelchior** [comment]
Closed by #4631
